herddb-core/src/main/java/herddb/sql/DDLSQLPlanner.java

```diff
@@ -886,17 +890,24 @@ private Statement buildAlterStatement(String defaultTableSpace, Alter alter) thr
                             }
                         }
                     }
+                    List<String> columnSpecs = decodeColumnSpecs(cl.getColumnSpecs());
                     int newType = sqlDataTypeToColumnType(
                             cl.getColDataType().getDataType(),
-                            cl.getColDataType().getArgumentsStringList()
+                            cl.getColDataType().getArgumentsStringList(),
+                            columnSpecs
                     );
 
                     if (oldColumn.type != newType) {
-                        throw new StatementExecutionException("cannot change datatype to " + cl.getColDataType().
-                                getDataType()
-                                + " for column " + columnName + " in table " + tableName + " in tablespace '" + tableSpace + "'");
+                        if (ColumnTypes.isNotNullToNullConversion(oldColumn.type, newType)) {
+                            // allow change from "STRING NOT NULL" to "STRING NULL"
+                        } else if (ColumnTypes.isNullToNotNullConversion(oldColumn.type, newType)) {
```

Comment:
For the null to NOT null use case what happens if the table already has null values for the columns?